ra-optimisation/linear-rao/src/test/java/com/farao_community/farao/linear_rao/LinearRaoOptimizerTest.java

```diff
@@ -0,0 +1,156 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.FileSystem;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Philippe Edwards {@literal <philippe.edwards at rte-france.com>}
+ */
+public class LinearRaoOptimizerTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinearRaoTest.class);
+
+    private LinearRaoOptimizer linearRaoOptimizer;
+    private ComputationManager computationManager;
+    private RaoParameters raoParameters;
+    private Crac crac;
+
+    @Before
+    public void setUp() {
+        FileSystem fileSystem = Jimfs.newFileSystem(Configuration.unix());
+        InMemoryPlatformConfig platformConfig = new InMemoryPlatformConfig(fileSystem);
+
+        computationManager = LocalComputationManager.getDefault();
+        raoParameters = RaoParameters.load(platformConfig);
+
+        LoadFlow.Runner loadFlowRunner = Mockito.mock(LoadFlow.Runner.class);
+        Mockito.when(loadFlowRunner.run(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any())).thenReturn(new LoadFlowResultImpl(true, Collections.emptyMap(), ""));
+        LoadFlowService.init(loadFlowRunner, computationManager);
+
+        crac = create();
+
+        Network network = Importers.loadNetwork(
+                "TestCase12Nodes.uct",
+                getClass().getResourceAsStream("/TestCase12Nodes.uct")
+        );
+
+        Map<State, SensitivityComputationResults> stateSensiMap = new HashMap<>();
+        Map<Cnec, Double> cnecMarginMap = new HashMap<>();
+        crac.getCnecs().forEach(cnec -> cnecMarginMap.put(cnec, 1.0));
+        Map<Cnec, Double> cnecMaxThresholdMap = new HashMap<>();
+        crac.getCnecs().forEach(cnec -> cnecMaxThresholdMap.put(cnec, 500.));
+        SystematicSensitivityAnalysisResult systematicSensitivityAnalysisResult = new SystematicSensitivityAnalysisResult(stateSensiMap, cnecMarginMap, cnecMaxThresholdMap);
+
+        LinearRaoProblem linearRaoProblemMock = Mockito.mock(LinearRaoProblem.class);
+
+        linearRaoOptimizer = new LinearRaoOptimizer(crac, network, systematicSensitivityAnalysisResult, computationManager, raoParameters, linearRaoProblemMock);
+    }
+
+    @Test
+    public void testUpdate() {
+        Map<State, SensitivityComputationResults> stateSensiMap = new HashMap<>();
+        Map<Cnec, Double> cnecMarginMap = new HashMap<>();
+        crac.getCnecs().forEach(cnec -> cnecMarginMap.put(cnec, 3.0));
+        Map<Cnec, Double> cnecMaxThresholdMap = new HashMap<>();
+        crac.getCnecs().forEach(cnec -> cnecMaxThresholdMap.put(cnec, 700.0));
+        SystematicSensitivityAnalysisResult systematicSensitivityAnalysisResult = new SystematicSensitivityAnalysisResult(stateSensiMap, cnecMarginMap, cnecMaxThresholdMap);
+        linearRaoOptimizer.update(systematicSensitivityAnalysisResult);
+        assertEquals(697, linearRaoOptimizer.getLinearRaoData().getReferenceFlow(crac.getCnecs().iterator().next()), 0.1);
+    }
```

Comment:
Is it really useful to create all this setup to just test that ?

Reply:
no, removed some of it